public void insert(Pedido pedido, EntityManager usuarioMngr) throws SystemException {
    Queue queue;
    QueueConnectionFactory queueConnectionFactory;
    QueueConnection connection = null;
    QueueSession session = null;
    QueueSender sender = null;
    try {
        Properties properties = new Properties();
        properties.put("java.naming.factory.initial", "org.jnp.interfaces.NamingContextFactory");
        properties.put("java.naming.factory.url.pkgs", "=org.jboss.naming:org.jnp.interfaces");
        properties.put("java.naming.provider.url", "localhost:1099");
        // logger.info("Dentro de initQueue");
        InitialContext ctx = new InitialContext(properties);
        queue = (Queue) ctx.lookup("queue/myQueue");
        queueConnectionFactory = (QueueConnectionFactory) ctx.lookup("UIL2ConnectionFactory");
        connection = queueConnectionFactory.createQueueConnection();
        session = connection.createQueueSession(false, QueueSession.AUTO_ACKNOWLEDGE);
        sender = session.createSender(queue);
        ObjectMessage estudianteObject = session.createObjectMessage();
        estudianteObject.setObject(pedido);
        sender.send(estudianteObject);
    } catch (NamingException e) {
        String msg = "An error was encountered trying to lookup an object from JNDI";
    // logger.error(msg);
    } catch (JMSException e) {
        // TODO Auto-generated catch block
        e.printStackTrace();
    }
}
